nsqd/protocol_v2.go

```diff
@@ -113,6 +113,9 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 	if client.Channel != nil {
 		client.Channel.RemoveClient(client.ID)
 	}
+	if err == io.EOF && atomic.LoadInt32(&client.State) == stateClosing {
```

Comment:
This is exactly what I had in mind, but I'm not sure `io.EOF` is the _only_ error possible in this state - if we're in `stateClosing` I think all errors should be ignored, right?